### ISSUE TITLE
Where does the magic numbers of 104., 116., 122. in input.py/Shape/subtract_mean come from ?

### ISSUE THREAD

**Issue by youkaichao** [WeiTang114/MVCNN-TensorFlow#9]
Where does the magic numbers of 104., 116., 122. in input.py/Shape/subtract_mean come from ? 
Does these numbers have special meaning?

**Comment by WeiTang114** [comment]
They are mean values of R/B/G relatively. They are from the mean values of ImageNet dataset. Since the pretrained model is converted from Caffe with [caffe-tensorflow](https://github.com/ethereon/caffe-tensorflow), if you are feeding a **natural image** to the pretrained model, you need to normalize it (i.e. subtract the mean values) to get desired output.

However, because we are feeding "white rendered images over black background (0's) ", in my experiments, there is no need to subtract the values.

**Comment by WeiTang114** [comment]
I'm closing this. Feel free to open this again if you find any further issue.
